docs/source/troubleshoot.mdx

```diff
@@ -0,0 +1,119 @@
+# Troubleshooting
+
+This guide aims to provide you the tools and knowledge required to navigate some common issues. If the suggestions listed 
+in this guide do not cover your such situation, please refer to the [Asking for Help](#asking-for-help) section to learn where to
+find help with your specific issue.
+
+## Uploading datasets to 🤗 Hub
+
+### Authentication issues
+
+If you are experiencing authentication issues when sharing a dataset on 🤗 Hub using [`Dataset.push_to_hub`] and a Hugging Face 
+access token:
+
+* Make sure that the Hugging Face token you're using to authenticate yourself is a token with **write** permission.
+* On OSX, it may help to clean up all the huggingface.co passwords on your keychain access, as well as reconfigure `git config --global credential.helper osxkeychain`, before using `huggingface-cli login`.
+
+Alternatively, you can use SSH keys to authenticate yourself - read more in the [🤗 Hub documentation](https://huggingface.co/docs/hub/security-git-ssh).
+
+### Lost connection on large dataset upload 
+
+When uploading large datasets to Hub, if the number of dataset shards is large, it can create too many commits for the Hub in a 
+short period. This will result in a connection error.
+The connection error can also be caused by a HTTP 500 error returned by AWS S3 bucket that Hub uses internally.
+In either situation, you can re-run [`Dataset.push_to_hub`] to proceed with the dataset upload. Hub will check the SHAs 
+of already uploaded shards to avoid reuploading them. 
+We are working on making upload process more robust to transient errors, so updating to the latest library version is 
+always a good idea.
+
+### `Too Many Requests`
+
+Uploading large datasets via `push_to_hub()` can result in an error: 
+
+```bash
+HfHubHTTPError: 429 Client Error: Too Many Requests for url: ...
+You have exceeded our hourly quotas for action: commit. We invite you to retry later.
+```
+
+If you encounter this issue, you need to upgrade the `datasets` library to the latest version (or at least `2.15.0`).
```

Comment:
Maybe we can group these sections into a "`push_to_hub` issues" section

Reply:
Done!